parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -30,79 +30,33 @@ pub use weights::WeightInfo;
 	Encode, Decode, CloneNoBound, PartialEqNoBound, RuntimeDebugNoBound, MaxEncodedLen, TypeInfo,
 )]
 #[scale_info(skip_type_params(M))]
-#[codec(mel_bound(AccountId: MaxEncodedLen))]
-pub struct EnqueuedMessage<AccountId, M: Get<u32>>
+#[codec(mel_bound(SourceId: MaxEncodedLen))]
+pub struct Message<SourceId, M: Get<u32>>
 where
-	AccountId: Encode + Decode + Clone + PartialEq + Debug + MaxEncodedLen + TypeInfo,
+	SourceId: Parameter + Member + MaxEncodedLen,
 {
-	account: AccountId,
-	message: Message<M>,
-}
-
-#[derive(
-	Encode, Decode, CloneNoBound, PartialEqNoBound, RuntimeDebugNoBound, MaxEncodedLen, TypeInfo,
-)]
-#[scale_info(skip_type_params(M))]
-#[codec(mel_bound())]
-pub struct Message<M: Get<u32>> {
-	/// Unique message ID
+	/// ID of source parachain
+	source_id: SourceId,
+	/// Unique nonce to prevent replaying messages
 	#[codec(compact)]
-	id: u64,
-	/// Target application on the Ethereum side.
-	target: H160,
+	nonce: u64,
 	/// Payload for target application.
 	payload: BoundedVec<u8, M>,
 }
 
-impl<M: Get<u32>> Into<Token> for Message<M> {
-	fn into(self) -> Token {
-		Token::Tuple(vec![
-			Token::Uint(self.id.into()),
-			Token::Address(self.target),
-			Token::Bytes(self.payload.to_vec()),
-		])
-	}
-}
-
-pub type EnqueuedMessageOf<T> =
-	EnqueuedMessage<<T as frame_system::Config>::AccountId, <T as Config>::MaxMessagePayloadSize>;
-
-/// Wire-format for committed messages
-#[derive(
-	Encode, Decode, CloneNoBound, PartialEqNoBound, RuntimeDebugNoBound, MaxEncodedLen, TypeInfo,
-)]
-#[scale_info(skip_type_params(M, N))]
-#[codec(mel_bound(AccountId: MaxEncodedLen))]
-pub struct MessageBundle<AccountId, M: Get<u32>, N: Get<u32>>
-where
-	AccountId: Encode + Decode + Clone + PartialEq + Debug + MaxEncodedLen + TypeInfo,
-{
-	account: AccountId,
-	/// Unique nonce to prevent replaying bundles
-	#[codec(compact)]
-	nonce: u64,
-	messages: BoundedVec<Message<M>, N>,
-}
-
-impl<AccountId, M: Get<u32>, N: Get<u32>> Into<Token> for MessageBundle<AccountId, M, N>
+impl<SourceId, M: Get<u32>> Into<Token> for Message<SourceId, M>
 where
-	AccountId: AsRef<[u8]> + Encode + Decode + Clone + PartialEq + Debug + MaxEncodedLen + TypeInfo,
+	SourceId: Decode + Parameter + Member + MaxEncodedLen, //+ TypeInfo,
 {
 	fn into(self) -> Token {
 		Token::Tuple(vec![
-			Token::FixedBytes(self.account.as_ref().into()),
+			Token::Bytes(self.source_id.encode()),
 			Token::Uint(self.nonce.into()),
-			Token::Array(self.messages.into_iter().map(|message| message.into()).collect()),
+			Token::Bytes(self.payload.to_vec()),
```

Comment:
is this `Message->Token` Convert still in use now? did not find referenced somewhere

Reply:
It's [used with `ethabi::encode`](https://github.com/Snowfork/snowbridge/pull/770/files/a343f4ae848411b053ea2b87f61abb92a19116f4#diff-3100992310c8525bbe27b87da7cc75b5a6a091d313a4fed5001406f24993bed9R225) to serialize the messages. `ethabi::encode` operates on Tokens, so we need to convert the message first.